apps/designer/app/shared/remix/create-trpc-remix-proxy.ts

```diff
@@ -0,0 +1,40 @@
+import { useFetcher } from "@remix-run/react";
+import type {
+  AnyMutationProcedure,
+  AnyRouter,
+  inferRouterInputs,
+  inferRouterOutputs,
+} from "@trpc/server";
+
+const proxy = (get: (method: string) => void) =>
+  new Proxy({}, { get: (_target, method: string) => get(method) });
```

Comment:
Why do this scary api with proxy? Could we just add separate useQuery, useMutation which  got trpc something with type info as argument and infer output api? Like relay does for fragments.

Reply:
So are we good with this after the discussion or are there serious concerns?